### PR TITLE
build: Drop per-host faketime wrappers in gitian-linux build

### PR DESCRIPTION
**Pull request by hebasto**
This PR is based on #19622, and the first two commits belong to #19622.

This is another step to removing `libfaketime` from gitian builds.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK on removing code, assuming that everything is still correct

File: contrib/gitian-descriptors/gitian-linux.yml [hunk 1]
```diff
@@ -12,14 +12,13 @@ packages:
...
 - "pkg-config"
 # Host dependencies.
-- "g++-8"
-- "g++-8-arm-linux-gnueabihf"
-- "g++-8-aarch64-linux-gnu"
-- "g++-8-riscv64-linux-gnu"
+- "g++"
+- "g++-arm-linux-gnueabihf"
+- "g++-aarch64-linux-gnu"
+- "g++-riscv64-linux-gnu"
```
Comment (dongcarl) [inline contrib/gitian-descriptors/gitian-linux.yml]: Let's not remove the explicit `-8` to be more explicit...
We can use `update-alternatives` in the script to change the default gcc/g++ I think

**Review by dongcarl — COMMENTED** [review]
Got some hashes you can post? :relaxed:

**Comment by hebasto** [comment]
@dongcarl 
> Got some hashes you can post?

```
1f6339377bcf2eb7f01233f730c8e36c07511c5a4be94a1bead96a9b2cb9aee0  bitcoin-55f3f74bf970-aarch64-linux-gnu-debug.tar.gz
8346d46076f47d39704ac7ab186a5bde4a9b549a8838ce925665f47b0f904dca  bitcoin-55f3f74bf970-aarch64-linux-gnu.tar.gz
c8ce9b2dd3c2d63cbab7d4757870999fb3900e0396d73b69f71b8c8a7c9f4caf  bitcoin-55f3f74bf970-arm-linux-gnueabihf-debug.tar.gz
d7a2aab88e81d9efe0b419944f1a97b8fb06f6aa08de4d96e1faefa13619b99d  bitcoin-55f3f74bf970-arm-linux-gnueabihf.tar.gz
7290e27027bd3e7a3fbbbb090d66838df401f80abd1868cb700152592892a6e5  bitcoin-55f3f74bf970-riscv64-linux-gnu-debug.tar.gz
3ec37df653eaaa504e2f1b02ab168093998e49e77d073978af8c80af11b54c47  bitcoin-55f3f74bf970-riscv64-linux-gnu.tar.gz
13c88bf1a3e38af78b5e6abebd9b2ed7c724d0dbd961339159f603ced0508df5  bitcoin-55f3f74bf970-x86_64-linux-gnu-debug.tar.gz
e65a8627729bfc578ca738bacca87ee1b7d692a9a50d022d563f182c83b14138  bitcoin-55f3f74bf970-x86_64-linux-gnu.tar.gz
2430de380ac7d943740ff1ce492a6a66e91da7071b4d381895ebf8400b4533ea  src/bitcoin-55f3f74bf970.tar.gz
```

**Comment by dongcarl** [comment]
@hebasto See: https://github.com/bitcoin/bitcoin/pull/19622#issuecomment-666648155

File: contrib/gitian-descriptors/gitian-linux.yml [hunk 2]
```diff
@@ -41,8 +28,7 @@ script: |
 
   WRAP_DIR=$HOME/wrapped
   HOSTS="x86_64-linux-gnu arm-linux-gnueabihf aarch64-linux-gnu riscv64-linux-gnu"
-  CONFIGFLAGS="--enable-glibc-back-compat --enable-reduce-exports --disable-bench --disable-gui-tests"
-  FAKETIME_HOST_PROGS="gcc g++"
+  CONFIGFLAGS="--enable-determinism --enable-glibc-back-compat --enable-reduce-exports --disable-bench --disable-gui-tests"
```
Comment (fanquake) [inline contrib/gitian-descriptors/gitian-linux.yml:31]: Passing `--enable-determinism` here isn't going to do anything, as that option currently only enables an additional linker argument when building for Windows.

Comment (hebasto) [reply contrib/gitian-descriptors/gitian-linux.yml:31]: :facepalm: indeed https://github.com/bitcoin/bitcoin/blob/a63a26f042134fa80356860c109edb25ac567552/configure.ac#L807-L811

Comment (hebasto) [reply contrib/gitian-descriptors/gitian-linux.yml:31]: Why do I observe deterministic builds without faketime wrappers?

**Pull request closed without merging**